params/hooks_libevm.go

```diff
@@ -19,6 +20,13 @@ import (
 	"golang.org/x/exp/maps"
 )
 
+type RulesExtra extras.Rules
```

Comment:
Why can't you register `extras.Rules` as the type?

Reply:
We can do this: https://github.com/ava-labs/coreth/pull/708/files however I prefer as is since:
- The call to GetExtra to construct the `accessibleState` for the precompiles cannot be type safe (also doesn't seem there's a path to return an error easily, namely `PrecompileOverride` doesn't have an error return).
- The behaviors remain in the package that registers the type, which makes more sense to me. This wasn't feasible with ChainConfig since there was already significant uses of the methods outside of libevm